tests/RobotFramework/tests/configuration/lock_file.robot

```diff
@@ -0,0 +1,36 @@
+*** Settings ***
+Resource    ../../resources/common.resource
+Library    ThinEdgeIO
+
+Test Tags    theme:configuration
+Suite Setup    Setup
+Test Teardown   Get Logs
+
+*** Test Cases ***
+
+Check lock file existence in default folder
+    [Documentation]    When deploying thin-edge components in single-process containers, 
+    ...    the lock file mechanism used by some components (e.g. tedge-agent, tedge-mapper) 
+    ...    do not makes sense as the container filesystem is isolated. 
+    ...    Having the lock file system just adds unnecessary dependencies on the /run/lock folder.
+    File Should Exist    /run/lock/tedge-agent.lock
+    File Should Exist    /run/lock/tedge-mapper-c8y.lock
+
+Switch off lock file creation
```

Comment:
How about adding another test that tries to start the same service twice and fails because there is already an instance of that service running?

Reply:
Though we can create another PR for this. The test was focussing on testing if the creation of lock files is configurable or not.